client/internal/clientcommon.go

```diff
@@ -191,12 +192,28 @@ func (c *ClientCommon) SetAgentDescription(descr *protobufs.AgentDescription) er
 // calcHashEffectiveConfig calculates and sets the Hash field from the rest of the
 // fields in the message.
 func calcHashEffectiveConfig(msg *protobufs.EffectiveConfig) {
+	cfgMap := msg.GetConfigMap().GetConfigMap()
```

Comment:
I think `msg.GetConfigMap()` can return nil and this needs to be checked before calling the other `GetConfigMap()`

Reply:
Because of the way protobuf code is generated, this chaining is safe.

```go
func (x *AgentConfigMap) GetConfigMap() map[string]*AgentConfigFile {
	if x != nil {
		return x.ConfigMap
	}
	return nil
}
```

Reply:
Oh, you're right. TBH, I wasn't aware this can be pulled with GoLang. Thanks!

Reply:
The generate proto code actually does a nil check on the pointer receiver so it'll just return nil and not panic.

https://github.com/open-telemetry/opamp-go/blob/main/protobufs/opamp.pb.go#L2463-L2468

Reply:
My first test case actually has a nil `AgentConfigMap` so it tests this case.